compiler/src/iree/compiler/Dialect/HAL/IR/HALTypes.cpp

```diff
@@ -308,6 +308,106 @@ SmallVector<IREE::HAL::DeviceTargetAttr, 4> DeviceTargetAttr::lookup(
   return {};  // No devices found; let caller decide what to do.
 }
 
+// Returns a set of all configuration attributes from all device targets.
+// Returns nullopt if any target is missing a configuration attribute.
+static std::optional<SmallVector<DictionaryAttr>> lookupConfigAttrs(
+    Operation *op) {
+  SmallVector<DictionaryAttr> configAttrs;
+  auto targetAttrs = IREE::HAL::DeviceTargetAttr::lookup(op);
+  if (targetAttrs.empty()) return std::nullopt;
+  for (auto targetAttr : targetAttrs) {
+    auto configAttr = targetAttr.getConfiguration();
+    if (!configAttr) return std::nullopt;
```

Comment:
Are we mostly expecting either all or none is set? (was just thinking that we are resizing an array which we don't know will even be needed, probably YAGNI)

Reply:
tweaked